src/Microsoft.DocAsCode.Common/Loggers/AggregatedPerformanceScope.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Linq;
+    using System.Runtime.InteropServices.ComTypes;
+    using System.Threading;
+
+    public sealed class AggregatedPerformanceScope : IDisposable
+    {
+        private readonly ConcurrentDictionary<string, Lazy<AggregatedPerformance>> _aggregatedPerformanceByPhase = new ConcurrentDictionary<string, Lazy<AggregatedPerformance>>();
+        private readonly LogLevel _logLevel = LogLevel.Verbose;
+
+        public AggregatedPerformanceScope(LogLevel? logLevel = null)
+        {
+            if (logLevel != null)
+            {
+                _logLevel = logLevel.Value;
+            }
+        }
+
+        public void Log(TimeSpan elapsedTime)
+        {
+            var phaseName = LoggerPhaseScope.GetPhaseName();
+            if (string.IsNullOrEmpty(phaseName))
+            {
+                return;
+            }
+
+            var aggregatedPerformanceByPhase = _aggregatedPerformanceByPhase.GetOrAdd(phaseName, _ => new Lazy<AggregatedPerformance>(() => new AggregatedPerformance())).Value;
+
+            aggregatedPerformanceByPhase.Log(elapsedTime.TotalMilliseconds);
+        }
+
+        public void Dispose()
+        {
+            foreach (var aggregatedPerformanceByPhase in _aggregatedPerformanceByPhase.OrderBy(kvp => kvp.Key))
+            {
+                var aggregatedPerformance = aggregatedPerformanceByPhase.Value.Value;
+                Logger.Log(_logLevel, $"Phase '{aggregatedPerformanceByPhase.Key}' runs {aggregatedPerformance.Occurrence} times with average time of {aggregatedPerformance.TotalTimeInMilliseconds / aggregatedPerformance.Occurrence} milliseconds.");
```

Comment:
>Occurrence [](start = 221, length = 10)

Although current code will not trigger, better to add a zero divider check. Can encapsulate the logic in `AggregatedPerformance.AverageTime`.